pkg/core/proxy/integrations/generic/decode.go

```diff
@@ -71,6 +71,17 @@ func decodeGeneric(ctx context.Context, logger *zap.Logger, reqBuf []byte, clien
 					logger.Debug("the genericRequests are:", zap.Any("h", string(genReq)))
 				}
 
+				if !opts.FallBackOnMiss {
+					_, err := clientConn.Write(([]byte{}))
+					if err != nil {
+						utils.LogError(logger, err, "failed to write empty response to the client")
+						errCh <- err
+						return
+					}
+					errCh <- nil
+					return
+				}
+
```

Comment:
Also add an info log that `no mock matched with the current request, hence connecting to the real service` along with the destination address in all the parsers.